src/phpDocumentor/Configuration/ConfigurationFactory.php

```diff
@@ -0,0 +1,332 @@
+                case 'version':
+                    $versions[(string) $value->attributes()->number] = $this->buildVersions($value);
+                    break;
+                case 'template':
+                    $template = $this->buildTemplate($value);
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        $phpdoc3Array = [
+            'phpdocumentor' => [
+                'paths'     => [
+                    'output' => (string) $phpDocumentor->paths->output,
+                    'cache'  => (string) $phpDocumentor->paths->cache,
+                ],
+                'versions'  => $versions,
+                'templates' => $template,
+            ]
+        ];
+
+        return $phpdoc3Array;
+    }
+
+    /**
+     * Builds the versions part of the array from the phpDocumentor3 configuration xml
+     *
+     * @param \SimpleXMLElement $version
+     *
+     * @return array
+     */
+    private function buildVersions(\SimpleXMLElement $version)
+    {
+        return [
+            'folder' => (string) $version->folder,
+            'api'    => [
+                'format'               => 'php',
+                'source'               => [
+                    'dsn'   => 'file://.',
+                    'paths' => [
+                        0 => 'src'
+                    ]
+                ],
+                'ignore'               => [
+                    'hidden'   => filter_var($version->api->ignore->attributes()->hidden, FILTER_VALIDATE_BOOLEAN),
+                    'symlinks' => filter_var($version->api->ignore->attributes()->symlinks, FILTER_VALIDATE_BOOLEAN),
+                    'paths'    => (array) $version->api->ignore->path,
+                ],
+                'extensions'           => [
+                    0 => 'php',
+                    1 => 'php3',
+                    2 => 'phtml'
+                ],
+                'visibility'           => 'public',
+                'default-package-name' => 'Default',
+                'markers'              => (array) $version->api->markers->children()->marker,
+            ],
+            'guide'  => [
+                'format' => (string) $version->guide->attributes()->format,
+                'source' => [
+                    'dsn'   => (string) $version->guide->source->attributes()->dsn,
+                    'paths' => (array) $version->guide->source->path,
+                ]
+            ]
+        ];
+    }
+
+    /**
+     * Builds the template part of the array from the phpDocumentor3 configuration xml
+     *
+     * @param \SimpleXMLElement $template
+     *
+     * @return array
+     */
+    private function buildTemplate(\SimpleXMLElement $template)
+    {
+        if (!$template) {
+            // Use default template if none is found in the configuration
+            return [
+                0 => [
+                    'name' => 'clean'
+                ],
+                1 => [
```

Comment:
This is not necessary; this was just an example on how you can use an external location to load an XML file